src/pip/_internal/network/session.py

```diff
@@ -218,8 +219,12 @@ def send(
         try:
             stats = os.stat(pathname)
         except OSError as exc:
+            # format the exception raised as a io.BytesIO object,
+            # to return a better error message:
             resp.status_code = 404
-            resp.raw = exc
+            error_message = f'"{type(exc).__name__}: {str(exc)}"'
+            resp.reason = error_message
+            resp.raw = io.BytesIO(error_message.encode("utf8"))
```

Comment:
Showing the path twice feels redundant to me. How about

```suggestion
            resp.reason = type(exc).__name__
            resp.raw = io.BytesIO(f"{resp.reason}: {exc}".encode("utf8"))
```